batch/batches/MoveCategoryEntries/KAsyncMoveCategoryEntries.class.php

```diff
@@ -187,60 +106,52 @@ private function moveEntries(KalturaBatchJob $job, KalturaMoveCategoryEntriesJob
 	{
 		$categoryEntryFilter = new KalturaCategoryEntryFilter();
 		$categoryEntryFilter->orderBy = KalturaCategoryEntryOrderBy::CREATED_AT_ASC;
-		$categoryEntryFilter->categoryIdEqual = $srcCategoryId;
-		
+		if($data->moveFromChildren)
+			$categoryEntryFilter->categoryFullIdsStartsWith = $data->destCategoryFullIds;
+		else
+			$categoryEntryFilter->categoryIdEqual = $srcCategoryId;
+
 		$categoryEntryPager = new KalturaFilterPager();
 		$categoryEntryPager->pageSize = 100;
+		$categoryEntryPager->pageIndex = 1;
+
 		if(KBatchBase::$taskConfig->params->pageSize)
 			$categoryEntryPager->pageSize = KBatchBase::$taskConfig->params->pageSize;
 			
-		if($data->lastMovedCategoryId == $srcCategoryId)
-			$categoryPager->pageIndex = $data->lastMovedCategoryEntryPageIndex;
-			
 		$movedEntries = 0;
 		$categoryEntriesList = KBatchBase::$kClient->categoryEntry->listAction($categoryEntryFilter, $categoryEntryPager);
```

Comment:
Move listAction into the do{} loop and you won't need to repeat it at the end of the loop.